xds/internal/balancer/outlierdetection/balancer.go

```diff
@@ -596,13 +606,35 @@ func (b *outlierDetectionBalancer) Target() string {
 // if the SubConn is not ejected.
 func (b *outlierDetectionBalancer) handleSubConnUpdate(u *scUpdate) {
 	scw := u.scw
-	scw.latestState = u.state
-	if !scw.ejected {
-		if scw.listener != nil {
-			b.childMu.Lock()
-			scw.listener(u.state)
-			b.childMu.Unlock()
-		}
+	scw.mu.Lock()
+	scw.healthListener = nil
+	scw.mu.Unlock()
+
+	if scw.listener == nil {
+		return
+	}
+
+	if !scw.healthListenerEnabled {
+		scw.latestDeleveredState = u.state
+	}
+
+	if scw.healthListenerEnabled || !scw.ejected {
```

Comment:
Why are we sending connectivity state updates for a subchannel that is ejected *but* has a health listener enabled? We used to *only* send connectivity state updates for subchannels that were not ejected.